packages/server/src/fhir/repo.ts

```diff
@@ -2446,23 +2453,21 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
     return this.context;
   }
 
-  close(): void {
+  [Symbol.dispose](): void {
     this.assertNotClosed();
-    if (this.transactionDepth > 0) {
-      // Bad state, remove connection from pool
-      getLogger().error('Closing Repository with active transaction');
-      this.releaseConnection(new Error('Closing Repository with active transaction'));
-    } else {
-      // Good state, return healthy connection to pool
-      this.releaseConnection();
+    if (this.disposable) {
+      if (this.transactionDepth > 0) {
+        // Bad state, remove connection from pool
+        getLogger().error('Closing Repository with active transaction');
+        this.releaseConnection(new Error('Closing Repository with active transaction'));
+      } else {
+        // Good state, return healthy connection to pool
+        this.releaseConnection();
+      }
     }
     this.closed = true;
   }
 
-  [Symbol.dispose](): void {
```

Comment:
Context (from offline conversation): `[Symbol.dispose]` is not automatically called unless using the `using` keyword, so although we are not currently taking advantage of that, we can standardize on `[Symbol.dispose]` as the "closer" for our various resources with explicit close functions.